defaults/main.yml

```diff
@@ -5,11 +5,28 @@ rhc_baseurl: null
 rhc_environments: []
 rhc_insights:
   ansible_host: null
+  autoconfig: true
   autoupdate: true
+  baseurl: "cert-api.access.redhat.com:443/r/insights"
```

Comment:
I would rather not have an explicit value like this, I would rather have a value which means "if `baseurl` is unset, use whatever is the default value built-in to insights client".

Reply:
I put the value in `defaults/main.yml` as I thought it is a good idea to set the default value from the `/etc/insights-client/insights-client.conf` as a default value for the role as well.

I can see that this could cause some issues if there are different defaults for different releases or when the default in insights-client change a change to the role would be necessary, too. I can change that.

Reply:
But what's the better approach in general? To put default values from client configs into `default/main.yml` or not? Is there some rule of thumb or does it depend on every single case?

When you can give me some advice I will consider it for future contributions.

Reply:
> But what's the better approach in general? To put default values from client configs into default/main.yml or not? Is there some rule of thumb or does it depend on every single case?

There is no hard and fast rule.  I haven't looked at https://github.com/redhat-cop/automation-good-practices to see if this is covered.  And maybe it is just my opinion, and others will probably disagree with me.

The problem with this particular variable is that now the default value is set it in two different places, and it puts a burden on the role maintainer to constantly check if the value has changed in `/etc/insights-client/insights-client.conf` in order to keep the role in sync with the insights default.  This is not good programming practice.  In addition, if the value in the role gets out of sync, then the failure case will be mysterious and hard to debug.

In the case of other values in defaults/main.yml, these reasons do not apply.

Reply:
I see your point and agree. I'm going to change the value in `defaults/main.yml` to `baseurl: null` and edit the corresponding task that it will only run when another value was provided.

Reply:
Fixed in commit [1600b96](https://github.com/linux-system-roles/rhc/pull/177/commits/1600b964e8a3ebc1772557532a5c80db72866d34).